### ISSUE TITLE
Text Inputs should support more types (email, url, tel, search)

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#524]
Text Inputs currently only support `text `and `password`s.

https://modus-web-components.trimble.com/?path=/story/user-inputs-text-input--default&args=includeSearchIcon:true;maxLength:1;required:true

They should really also support `email`, `url`, `tel` and `search `- these are useful for presenting user with an optimised virtual keyboard on touch screen devices and can aid accessibility too.

REF: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types

**Comment by msankaran0712** [comment]
@coliff Do we need to keep this issue?

**Comment by coliff** [comment]
yes, I think it's important - helps with usability and a11y for forms. It should be quite easy to do. I might even see if I can do it myself :-)
